app/src/main/java/com/example/keniphhomemaintenance/maintenance_screen/MaintenanceItemDialog.kt

```diff
@@ -0,0 +1,130 @@
+                    placeholder = R.string.maintenance_item_dialog_task_name_field_placeholder,
+                    focusAction = FocusOptions.Down
+                )
+
+                Spacer(modifier = Modifier.padding(MAINTENANCE_ITEM_DIALOG_PADDING_BETWEEN_TEXT_FIELDS))
+
+                MaintenanceTextField(
+                    fieldValue = dueDate,
+                    fieldLabel = R.string.due_date_field_label,
+                    placeholder = R.string.maintenance_item_dialog_due_date_field_placeholder,
+                    focusAction = FocusOptions.Down
+                )
+
+                Spacer(modifier = Modifier.padding(MAINTENANCE_ITEM_DIALOG_PADDING_BETWEEN_TEXT_FIELDS))
+
+                MaintenanceTextField(
+                    fieldValue = location,
+                    fieldLabel = R.string.location_field_label,
+                    placeholder = R.string.maintenance_item_dialog_location_field_placeholder,
+                    focusAction = FocusOptions.Clear
+                )
+
+                Spacer(modifier = Modifier.padding(MAINTENANCE_ITEM_DIALOG_LOCATION_FIELD_BOTTOM_PADDING))
+
+                Row(
+                    modifier = Modifier
+                        .fillMaxWidth(),
+                    verticalAlignment = Alignment.CenterVertically,
+                    horizontalArrangement = Arrangement.SpaceEvenly
+                ) {
+
+                        ExtendedFloatingActionButton(
+                            onClick = {
+                                maintenanceItemDialogEFABOnClick(
+                                    isDialogOpen = isDialogOpen,
+                                    itemTitle = itemTitle,
+                                    dueDate = dueDate,
+                                    location = location
+                                )
+                            },
+                            text = { Text(text = stringResource(id = R.string.maintenance_item_dialog_cancel_button_text)) }
+                        )
+
+                    ExtendedFloatingActionButton(
+                        onClick = {
+                            maintenanceViewModel.addMaintenanceItem(itemTitle.value, dueDate.value, location.value)
+                            maintenanceItemDialogEFABOnClick(
+                                isDialogOpen = isDialogOpen,
+                                itemTitle = itemTitle,
+                                dueDate = dueDate,
+                                location = location
+                            )
+                        },
+                        text = { Text(text = stringResource(id = R.string.maintenance_item_dialog_add_item_button_text)) }
+                    )
+
+                }
+            }
+        }
+    }
+}
+
+fun maintenanceItemDialogEFABOnClick(isDialogOpen: MutableState<Boolean>, itemTitle: MutableState<String>, dueDate: MutableState<String>, location: MutableState<String>) {
```

Comment:
Couple nitpicks
- The name of this function ties it to the UI element ("EFAB"onClick). We should avoid doing so in case we change from an EFAB to another type of button in the future. 
- What does this function do? What does it control? Something like "onMaintenanceItemDialogClose" is probably a good call. This function describes the set of actions that should _always_ happen when the MaintenanceItemDialog is closed. We're tying the function name to the UI a little bit by including "Dialog", but if we change this from a Dialog to another screen, we'll be changing a bunch of code anyways, so I don't know if there's a way around that.